### PR TITLE
Re-run yarn install process when relevant env var configuration changes

### LINKED ISSUE #128: Buildpack reusing module layer even when the value of env vars change
It looks like the buildpack does not consider env variables in its decision to reuse the module layer or not. For e.g., a user should be able to pass `NODE_ENV=development|production` and influence the actions of the yarn-install command. If the env vars are not considered in the decision to reuse layer, `yarn install` wouldn't be run again with the new env vars context.

Logs while using [this app](https://github.com/paketo-buildpacks/nodejs/tree/v0.0.8/integration/testdata/yarn) with the nodejs metabuildpack. See relevant "Yarn Install" section. The buildpack is `Reusing cached layer /layers/paketo-buildpacks_yarn-install/modules` though the value of `myenv` has changed in the second build.

```
# First Build
pack build myapp --env myenv=foo -b gcr.io/paketo-buildpacks/nodejs

===> DETECTING
[detector] paketo-buildpacks/node-engine  0.1.2
[detector] paketo-buildpacks/yarn         0.0.2
[detector] paketo-buildpacks/yarn-install 0.2.0
[detector] paketo-buildpacks/yarn-start   0.0.2
===> ANALYZING
[analyzer] Previous image with name "index.docker.io/library/myapp:latest" not found
[analyzer] Restoring metadata for "paketo-buildpacks/yarn:yarn" from cache
===> RESTORING
[restorer] Restoring data for "paketo-buildpacks/yarn:yarn" from cache
===> BUILDING
[builder] Paketo Node Engine Buildpack 0.1.2
[builder]   Resolving Node Engine version
[builder]     Candidate version sources (in priority order):
[builder]                 -> ""
[builder]       <unknown> -> "*"
[builder]
[builder]     Selected Node Engine version (using ): 10.22.1
[builder]
[builder]   Executing build process
[builder]     Installing Node Engine 10.22.1
[builder]       Completed in 1.461s
[builder]
[builder]   Configuring environment
[builder]     NODE_ENV     -> "production"
[builder]     NODE_HOME    -> "/layers/paketo-buildpacks_node-engine/node"
[builder]     NODE_VERBOSE -> "false"
[builder]
[builder]     Writing profile.d/0_memory_available.sh
[builder]       Calculates available memory based on container limits at launch time.
[builder]       Made available in the MEMORY_AVAILABLE environment variable.
[builder]
[builder] Paketo Yarn Buildpack 0.0.2
[builder]   Reusing cached layer /layers/paketo-buildpacks_yarn/yarn
[builder]
[builder] Paketo Yarn Install Buildpack 0.2.0
[builder]   Resolving installation process
[builder]     Process inputs:
[builder]       yarn.lock -> Found
[builder]
[builder]     Selected default build process: 'yarn install'
[builder]
[builder]   Executing build process
[builder]     Running yarn install --ignore-engines --frozen-lockfile --modules-folder /layers/paketo-buildpacks_yarn-install/modules/node_modules
[builder]       Completed in 1.046s
[builder]
[builder]   Configuring environment
[builder]     PATH -> "$PATH:/layers/paketo-buildpacks_yarn-install/modules/node_modules/.bin"
[builder] Paketo Yarn Start Buildpack 0.0.2
[builder]   Assigning launch processes
[builder]     web: node server.js
[builder]
===> EXPORTING
[exporter] Adding layer 'paketo-buildpacks/node-engine:node'
[exporter] Adding layer 'paketo-buildpacks/yarn-install:modules'
[exporter] Adding 1/1 app layer(s)
[exporter] Adding layer 'launcher'
[exporter] Adding layer 'config'
[exporter] Adding layer 'process-types'
[exporter] Adding label 'io.buildpacks.lifecycle.metadata'
[exporter] Adding label 'io.buildpacks.build.metadata'
[exporter] Adding label 'io.buildpacks.project.metadata'
[exporter] Setting default process type 'web'
[exporter] *** Images (a11747b628bc):
[exporter]       index.docker.io/library/myapp:latest
[exporter] Reusing cache layer 'paketo-buildpacks/node-engine:node'
[exporter] Reusing cache layer 'paketo-buildpacks/yarn:yarn'
[exporter] Reusing cache layer 'paketo-buildpacks/yarn-install:modules'
Successfully built image myapp

# Now build again with a different value for the env var
$ pack build myapp --env myenv=bar -b gcr.io/paketo-buildpacks/nodejs

===> DETECTING
[detector] paketo-buildpacks/node-engine  0.1.2
[detector] paketo-buildpacks/yarn         0.0.2
[detector] paketo-buildpacks/yarn-install 0.2.0
[detector] paketo-buildpacks/yarn-start   0.0.2
===> ANALYZING
[analyzer] Restoring metadata for "paketo-buildpacks/node-engine:node" from app image
[analyzer] Restoring metadata for "paketo-buildpacks/yarn:yarn" from cache
[analyzer] Restoring metadata for "paketo-buildpacks/yarn-install:modules" from app image
===> RESTORING
[restorer] Restoring data for "paketo-buildpacks/node-engine:node" from cache
[restorer] Restoring data for "paketo-buildpacks/yarn:yarn" from cache
[restorer] Restoring data for "paketo-buildpacks/yarn-install:modules" from cache
===> BUILDING
[builder] Paketo Node Engine Buildpack 0.1.2
[builder]   Resolving Node Engine version
[builder]     Candidate version sources (in priority order):
[builder]                 -> ""
[builder]       <unknown> -> "*"
[builder]
[builder]     Selected Node Engine version (using ): 10.22.1
[builder]
[builder]   Reusing cached layer /layers/paketo-buildpacks_node-engine/node
[builder]
[builder] Paketo Yarn Buildpack 0.0.2
[builder]   Reusing cached layer /layers/paketo-buildpacks_yarn/yarn
[builder]
[builder] Paketo Yarn Install Buildpack 0.2.0
[builder]   Resolving installation process
[builder]     Process inputs:
[builder]       yarn.lock -> Found
[builder]
[builder]     Selected default build process: 'yarn install'
[builder]
[builder]   Reusing cached layer /layers/paketo-buildpacks_yarn-install/modules
[builder] Paketo Yarn Start Buildpack 0.0.2
[builder]   Assigning launch processes
[builder]     web: node server.js
[builder]
===> EXPORTING
[exporter] Reusing layer 'paketo-buildpacks/node-engine:node'
[exporter] Reusing layer 'paketo-buildpacks/yarn-install:modules'
[exporter] Reusing 1/1 app layer(s)
[exporter] Reusing layer 'launcher'
[exporter] Reusing layer 'config'
[exporter] Reusing layer 'process-types'
[exporter] Adding label 'io.buildpacks.lifecycle.metadata'
[exporter] Adding label 'io.buildpacks.build.metadata'
[exporter] Adding label 'io.buildpacks.project.metadata'
[exporter] Setting default process type 'web'
[exporter] *** Images (a11747b628bc):
[exporter]       index.docker.io/library/myapp:latest
[exporter] Reusing cache layer 'paketo-buildpacks/node-engine:node'
[exporter] Reusing cache layer 'paketo-buildpacks/yarn:yarn'
[exporter] Reusing cache layer 'paketo-buildpacks/yarn-install:modules'
Successfully built image myapp
```

### PR DESCRIPTION
**Pull request by fg-j**
Signed-off-by: Emily Johnson <emjohnson@vmware.com>

## Summary

Resolves #128  

## Checklist

* [x] I have viewed, signed, and submitted the Contributor License Agreement.
* [x] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [x] I have added an integration test, if necessary.
* [ ] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.

### CODE REVIEW

File: install_process_test.go [hunk 1]
```diff
@@ -28,16 +28,27 @@ func testInstallProcess(t *testing.T, context spec.G, it spec.S) {
 			executable     *fakes.Executable
 			installProcess yarninstall.YarnInstallProcess
 			summer         *fakes.Summer
+			executions     []pexec.Execution
```
Comment (emmjohnson) [inline install_process_test.go:31]: This doesn't need to be an array since we just make one call

File: install_process.go [hunk 2]
```diff
@@ -52,9 +53,30 @@ func (ip YarnInstallProcess) ShouldRun(workingDir string, metadata map[string]in
...
+		return true, "", fmt.Errorf("failed to execute yarn config output:\n%s\nerror: %s", buffer.String(), err)
+	}
+
+	configFile, err := createTempFileAndWriteContents(buffer.Bytes(), "config-file")
+	if err != nil {
+		return true, "", err
+	}
+
+	nodeEnv := os.Getenv("NODE_ENV")
+	nodeEnvFile, err := createTempFileAndWriteContents([]byte(nodeEnv), "node-env")
```
Comment (arjun024) [inline install_process.go:72]: If the purpose is to simply calculate the checksum, could you append this data to the same file to avoid redundant I/O?

**Review by arjun024 — CHANGES_REQUESTED** [review]
Could you please take a look at the comment I left? Other than that, looks great to me!

File: install_process.go [hunk 3]
```diff
@@ -63,18 +63,23 @@ func (ip YarnInstallProcess) ShouldRun(workingDir string, metadata map[string]in
 		return true, "", fmt.Errorf("failed to execute yarn config output:\n%s\nerror: %s", buffer.String(), err)
 	}
 
-	configFile, err := createTempFileAndWriteContents(buffer.Bytes(), "config-file")
+	nodeEnv := os.Getenv("NODE_ENV")
+	_, err = buffer.WriteString(nodeEnv)
 	if err != nil {
```
Comment (arjun024) [inline install_process.go:68]: Looks like `(*Buffer) WriteString` always returns nil. Could you please remove the error check?

**Review by arjun024 — CHANGES_REQUESTED** [review]
Alright final requested change :) LGTM otherwise

**Review by arjun024 — APPROVED** [review]

**Pull request merged**